File: cpp/questions/minBitwiseArray.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        int n = nums.size();
        std::vector<int> ans(n, -1);
        for (int p=0; p<n; p++) {
            if (nums[p] & 1) {
                int tmp = nums[p];
                int cnt = 0;
                while (tmp & 1) {
                    cnt++;
                    tmp >>= 1;
                }
                ans[p] = nums[p] & ~(1<<(cnt-1));
            }
        }
        return ans;
    }
};

File: python/questions/minBitwiseArray.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [-1 for _ in range(n)]
        for p in range(n):
            if nums[p] & 1:
                tmp = nums[p]
                cnt = 0
                while tmp & 1:
                    cnt += 1
                    tmp >>= 1
                ans[p] = nums[p] & ~(1<<(cnt-1))
        return ans
